File: itzel_fe/src/app/modelos/contabilidad/cuentas.ts
import { Niveles } from './niveles';

export class Cuentas {
  idcuenta?: number;
  codcue?: string;
  nomcue?: string;
  movcue?: Boolean;
  idnivel?: Niveles;
  grucue?: string;
  asodebe?: string;
  asohaber?: string;
  tiptran?: number;
  sigef?: Boolean;
  grubalances?: number;
  grufluefec?: number;
  resultcostos?: number;
  balancostos?: number;
  usucrea?: number;
  feccrea?: Date;
  usumodi?: number;
  fecmodi?: Date;
}

File: itzel_be/src/main/java/com/itzel/modelo/contabilidad/Cuentas.java
package com.itzel.modelo.contabilidad;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name="cuentas")
public class Cuentas {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idcuenta;
    private String codcue;
    private String nomcue;
    private Boolean movcue;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idnivel")
    private Niveles nivel;
    private String grucue;
    private String asodebe;
    private String asohaber;
    private Long tiptran;
    private Boolean sigef;
    private Long grubalances;
    private Long grufluefec;
    private Long resulcostos;
    private Long balancostos;
    private Long usucrea;
    private LocalDate feccrea;
    private Long usumodi;
    private LocalDate fecmodi;

}
